components/brave_wallet/browser/cardano/cardano_transaction_unittest.cc

```diff
@@ -0,0 +1,247 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/cardano/cardano_transaction.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/strings/string_number_conversions.h"
+#include "base/test/values_test_util.h"
+#include "brave/components/brave_wallet/browser/cardano/cardano_rpc_schema.h"
+#include "brave/components/json/json_helper.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using testing::_;
+
+namespace brave_wallet {
+namespace {
+constexpr char kTxid1[] =
+    "aa388f50b725767653e150ad8990ec11a2146d75acafbe492af08213849fe2c5";
+constexpr char kTxid2[] =
+    "bd1c9cfb126a519f3ee593bbbba41a0f9d55b4d267e9483673a848242bc5c2be";
+constexpr char kAddress1[] =
+    "addr1qx2fxv2umyhttkxyxp8x0dlpdt3k6cwng5pxj3jhsydzer3n0d3vllmyqwsx5wktcd8cc"
+    "3sq835lu7drv2xwl2wywfgse35a3x";
+constexpr char kAddress2[] =
+    "addr1q9cwvremt6n320s2e3agq0jyq82yhrk3htsu0w426xnz5us70z4w0jgvcdkkynmm8wmds"
+    "66jd9kusnjfpu6raw5fqp0sr07p5w";
+
+}  // namespace
+
+TEST(CardanoTransaction, Outpoint_Value) {
+  CardanoTransaction::Outpoint outpoint;
+  outpoint.index = 123;
+  base::HexStringToSpan(kTxid1, outpoint.txid);
+
+  auto parsed = outpoint.FromValue(outpoint.ToValue());
+  ASSERT_TRUE(parsed);
+  EXPECT_EQ(*parsed, outpoint);
+  EXPECT_EQ(parsed->index, outpoint.index);
+  EXPECT_EQ(parsed->txid, outpoint.txid);
+}
+
+TEST(CardanoTransaction, TxInput_Value) {
+  CardanoTransaction::TxInput input;
+  input.utxo_address = *CardanoAddress::FromString(kAddress1);
+  input.utxo_outpoint.index = 123;
+  base::HexStringToSpan(kTxid1, input.utxo_outpoint.txid);
+  input.utxo_value = 555666777;
+
+  auto parsed = input.FromValue(input.ToValue());
+  ASSERT_TRUE(parsed);
+  EXPECT_EQ(*parsed, input);
+  EXPECT_EQ(parsed->utxo_address, input.utxo_address);
+  EXPECT_EQ(parsed->utxo_outpoint, input.utxo_outpoint);
+}
+
+TEST(CardanoTransaction, TxInput_FromRpcUtxo) {
+  const std::string rpc_utxo_json = R"(
```

Comment:
This is not sufficient testing for the json parsing. Unit tests need to cover edge cases including dealing with malformed json and json that doesn't match our expectations in terms of fields and values. Only testing the expected correct case is never sufficient.

Reply:
removed json parsing from here